easybuild/easyblocks/i/ipp.py

```diff
@@ -54,24 +69,52 @@ def install_step(self):
                 'license_file_name': LICENSE_FILE_NAME_2012,
             }
 
-        super(EB_ipp, self).install_step(silent_cfg_names_map=silent_cfg_names_map)
+        # in case of IPP 9.x, we have to specify ARCH_SELECTED in silent.cfg
+        if LooseVersion(self.version) >= LooseVersion('9.0'):
+            silent_cfg_extras = {
+                'ARCH_SELECTED': self.arch.upper()
+            }
+
+        super(EB_ipp, self).install_step(silent_cfg_names_map=silent_cfg_names_map, silent_cfg_extras=silent_cfg_extras)
 
     def sanity_check_step(self):
         """Custom sanity check paths for IPP."""
 
         if LooseVersion(self.version) < LooseVersion('8.0'):
-            dirs = ["compiler/lib/intel64", "ipp/bin", "ipp/include",
-                    "ipp/interfaces/data-compression", "ipp/tools/intel64"]
+            dirs = ['compiler/lib/intel64', 'ipp/bin', 'ipp/include',
+                    'ipp/interfaces/data-compression', 'ipp/tools/intel64']
+        elif LooseVersion(self.version) >= LooseVersion('9.0'):
+            dirs = ['ipp/bin', 'ipp/include', 'ipp/tools/intel64']
         else:
-            dirs = ["composerxe/lib/intel64", "ipp/bin", "ipp/include",
-                    "ipp/tools/intel64"]
+            dirs = ['composerxe/lib/intel64', 'ipp/bin', 'ipp/include',
+                    'ipp/tools/intel64']
+
+        ipp_libs = ['cc', 'ch', 'core', 'cv', 'dc', 'i', 's', 'vm']
+        if LooseVersion(self.version) < LooseVersion('9.0'):
+            ipp_libs.extend(['ac', 'di', 'j', 'm', 'r', 'sc', 'vc'])
 
         custom_paths = {
-                        'files': ["ipp/lib/intel64/libipp%s" % y
-                                   for x in ["ac", "cc", "ch", "core", "cv", "dc", "di",
-                                             "i", "j", "m", "r", "s", "sc", "vc", "vm"]
-                                   for y in ["%s.a" % x, "%s.so" % x]],
-                        'dirs': dirs
-                       }
+            'files': ['ipp/lib/intel64/libipp%s' % y for x in ipp_libs for y in ['%s.a' % x, '%s.so' % x]],
+            'dirs': dirs,
+        }
 
         super(EB_ipp, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_req_guess(self):
+        """
+        A dictionary of possible directories to look for
+        """
+        guesses = super(EB_ipp, self).make_module_req_guess()
+
+        if LooseVersion(self.version) >= LooseVersion('9.0'):
+            lib_path = os.path.join('lib', self.arch)
+            include_path = 'ipp/include'
+
+            guesses.update({
+                'LD_LIBRARY_PATH': [lib_path],
```

Comment:
good point, but it doesn't matter here

* `lib` is not useful, `lib/intel64` is (which is `lib_path` here)
* `include` doesn't exist, `ipp/include` does

so, the paths in the original `guesses` are useless/wrong